### PR TITLE
Updated function inObservation : R code, documentation and tests added.

### PR DESCRIPTION
**Pull request by KimLopezGuell**

### CODE REVIEW

File: R/inObservation.R [hunk 1]
```diff
@@ -14,39 +14,165 @@
...
-          .data$cohort_start_date <= .data$observation_period_end_date,
-        TRUE,
-        FALSE
+
+  ## check for standard types of user error
+
+  errorMessage <- checkmate::makeAssertCollection()
+
+  xCheck <- inherits(x, "tbl_dbi")
+  checkmate::assertTRUE(xCheck, add = errorMessage)
```
Comment (catalamarti) [inline R/inObservation.R:52]: I think this line is not necessary

File: R/inObservation.R [hunk 2]
```diff
@@ -14,39 +14,165 @@
...
+  if (!isTRUE(xCheck)) {
+    errorMessage$push(
+      "- x is not a table"
+    )
+  }
+
+  cdmCheck <- inherits(cdm, "cdm_reference")
+  checkmate::assertTRUE(cdmCheck,
```
Comment (catalamarti) [inline R/inObservation.R:60]: same here

File: R/inObservation.R [hunk 3]
```diff
@@ -14,39 +14,165 @@
...
+    errorMessage$push(
+      "- cdm must be a CDMConnector CDM reference object"
+    )
+  }
+
+  checkmate::reportAssertions(collection = errorMessage)
+
+  errorMessage <- checkmate::makeAssertCollection()
+
+  cdmObsPeriodCheck <- inherits(cdm$observation_period, "tbl_dbi")
```
Comment (catalamarti) [inline R/inObservation.R:73]: you have to check first if observation period is contained in cdm

File: R/inObservation.R [hunk 4]
```diff
@@ -14,39 +14,165 @@
...
+  if (!isTRUE(cdmObsPeriodCheck)) {
+    errorMessage$push(
+      "- table `observation_period` is not found in cdm"
+    )
+  }
+  checkmate::reportAssertions(collection = errorMessage)
+
+  errorMessage <- checkmate::makeAssertCollection()
+
+  checkmate::assertCharacter(observationAt,
```
Comment (catalamarti) [inline R/inObservation.R:84]: how to handle if length is not 1?

File: R/inObservation.R [hunk 5]
```diff
@@ -14,39 +14,165 @@
...
+  checkmate::assertTRUE(column1Check,
+                        add = errorMessage
+  )
+  if (!isTRUE(column1Check)) {
+    errorMessage$push(
+      "- `observationAt` is not a column of x"
+    )
+  }
+
+  column2Check <- "subject_id" %in% colnames(x)
```
Comment (catalamarti) [inline R/inObservation.R:98]: It would be nice that you can also work with a table with person_id instead of subject_id

File: R/inObservation.R [hunk 6]
```diff
@@ -14,39 +14,165 @@
...
+  checkmate::assertTRUE(column2Check,
+                        add = errorMessage
+  )
+  if (!isTRUE(column2Check)) {
+    errorMessage$push(
+      "- `subject_id` is not a column of x"
+    )
+  }
+
+  column3Check <- "person_id" %in% colnames(cdm$observation_period)
```
Comment (catalamarti) [inline R/inObservation.R:108]: I would not tdo this check as it is out of scope of the function, but happy if you want to include it, same for the subsequent ones.

File: R/inObservation.R [hunk 7]
```diff
@@ -14,39 +14,165 @@
...
+    errorMessage$push(
+      "- `observation_period_end_date` is not a column of cdm$observation_period"
+    )
+  }
+
+  checkmate::assertCharacter(name,
+                             add = errorMessage,
+  )
+
+  checkmate::assert_logical(compute,
```
Comment (catalamarti) [inline R/inObservation.R:142]: len = 1

File: R/inObservation.R [hunk 8]
```diff
@@ -14,39 +14,165 @@
...
+  checkmate::assertTRUE(column5Check,
+                        add = errorMessage
+  )
+  if (!isTRUE(column5Check)) {
+    errorMessage$push(
+      "- `observation_period_end_date` is not a column of cdm$observation_period"
+    )
+  }
+
+  checkmate::assertCharacter(name,
```
Comment (catalamarti) [inline R/inObservation.R:138]: len = 1?

**Review by catalamarti — CHANGES_REQUESTED** [review]

**Comment by KimLopezGuell** [comment]
Alternative pull request added that does not have problems in dependencies. Therefore this branch can be deleted.

**Pull request closed without merging**